offa.csl

```diff
@@ -0,0 +1,438 @@
+          <group delimiter=" ">
+            <choose>
+              <if type="thesis" match="any">
+                <text variable="genre" suffix=" "/>
+                <text variable="publisher"/>
+              </if>
+              <else>
+                <choose>
+                  <if match="any" variable="original-date original-publisher-place">
+                    <text variable="original-publisher-place"/>
+                    <date date-parts="year" form="text" variable="original-date" prefix=" "/>
+                    <text value="; "/>
+                  </if>
+                </choose>
+                <text variable="publisher-place"/>
+              </else>
+            </choose>
+            <text macro="date"/>
+          </group>
+          <text macro="edition"/>
+        </group>
+      </if>
+    </choose>
+  </macro>
+  <macro name="journal-date">
+    <choose>
+      <if type="article-journal">
+        <text macro="date" prefix=", "/>
+      </if>
+    </choose>
+  </macro>
+  <macro name="day-month">
+    <date variable="issued">
+      <date-part name="month"/>
+      <date-part name="day" prefix=" "/>
+    </date>
+  </macro>
+  <macro name="collection-title">
+    <text variable="container-title" prefix=" "/>
+    <text variable="volume" prefix=" "/>
+  </macro>
+  <macro name="event">
+    <text term="presented at" suffix=" "/>
+    <date form="text" variable="issued" suffix=", "/>
+    <text variable="event-place" suffix=". "/>
+    <text variable="event"/>
+  </macro>
+  <macro name="issue">
+    <choose>
+      <if type="speech" match="any">
+        <text macro="event" prefix=" "/>
+      </if>
+      <else-if type="paper-conference" match="any">
+        <text variable="event" prefix=" "/>
+        <date form="text" variable="event-date" prefix=" "/>
+      </else-if>
+      <else-if type="article-newspaper article-magazine" match="any">
+        <text macro="day-month" prefix=", "/>
+      </else-if>
+    </choose>
+  </macro>
+  <macro name="series">
+    <choose>
+      <if match="any" variable="collection-title">
+        <text variable="collection-title" prefix=". "/>
+        <text variable="collection-number" prefix=" "/>
+      </if>
+    </choose>
+    <choose>
+      <if type="paper-conference" match="any">
+        <text variable="volume" prefix=" "/>
+      </if>
+    </choose>
+  </macro>
+  <macro name="url">
+    <choose>
+      <if match="any" variable="URL">
+        <group>
+          <text variable="URL" text-case="lowercase" prefix=". "/>
+          <choose>
+            <if match="none" is-uncertain-date="accessed">
+              <group>
+                <text term="accessed" prefix=" ("/>
+                <date form="numeric" variable="accessed" suffix=")"/>
+              </group>
+            </if>
+          </choose>
+        </group>
+      </if>
+    </choose>
+  </macro>
+  <macro name="doi">
+    <choose>
+      <if match="any" variable="DOI">
```

Comment:
you test for this twice (here and in the bibliography) but don't actually need to at all: the group acts as a test and will only print if a DOI exists

Reply:
Thanks for clarifying! Corrected. Also corrected in style journal-of-neolithic-archaeology, as the offa-style derives from the latter.